### ISSUE TITLE
Request: Provide nearest neighbors for training data

### ISSUE THREAD

**Issue by dataplayer12** [erikbern/ann-benchmarks#51]
For datasets like SIFT 1M, it would be **great** if nearest neighbors were provided for training data as well. From the perspective of computational cost, the test data is very small and users can create its metrics themsleves. But if you could provide the NNs for 1M training points that would help those who want to use the larger datasets.

**Comment by erikbern** [comment]
Feel free to submit a pull request! Should be quite easy – here's the relevant code: https://github.com/erikbern/ann-benchmarks/blob/master/ann_benchmarks/datasets.py#L36

I can re-run and re-upload datasets

**Comment by dataplayer12** [comment]
Thanks for pointing me to the right place. I have just submitted a PR.

**Comment by dataplayer12** [comment]
Small change. On line 69, please replace `res.remove(x)` with `res.remove((x,0.))`. I forgot this was a list of tuples.

**Comment by yurymalkov** [comment]
@dataplayer12 Can you explain the significance of testing on the "train" data? 
I think any hash table with precomputed nearest neighbors for each dataset point can solve this task perfectly. 
The test results on train data will be significantly different in terms of both speed and accuracy. i.e. k-nn graph algorithms store precomputed nearest neighbors for each of the dataset point and show way better performance on "train" data.

**Comment by dataplayer12** [comment]
I am not sure that the division of 'train' and 'test' data is very strict here. Unless one is using a neural network or some other supervised learning technique, hashing is mostly an unsupervised task. Having 1Mx100 nearest neighbors for training dataset will make the dataset more useful for the community by allowing it to be used in contexts other than the framework and evaluation method defined in this repo.  Hope that helps.

**Comment by erikbern** [comment]
closing this – i remember trying at some point to compute nearest neighbors for the training set but it's brutally slow

**Comment by dataplayer12** [comment]
Can't argue about it being slow. Thank you for trying.
